src/include/OpenImageIO/fmath.h

```diff
@@ -1742,4 +1753,181 @@ interpolate_linear (float x, array_view_strided<const float> y)
 
 
 
+
+////////////////////////////////////////////////////////////////////////////
+// CUDA device functions & intrinsics
+//
+// For now, the selection of 'fast' instrinsics is controlled via compiler
+// flags.
+
+#ifdef __CUDACC__
+__device__ inline float fast_sin (float x) {
```

Comment:
I'm wondering if this should either be right next to the other definition of fast_sin, or if there should be ONE fast_sin and the #ifdef should be inside its body?

I'm not sure I have a strong opinion either way (or this way), but I'm trying to think ahead about the long-term maintainability of this file. If we have multiple definitions of the same function for different platforms, I think in general that we want as little textural separation between them as we can do practically. (That said, I can also argue for this approach of just succinctly having all of the Cuda definitions in one compact place. So I'm not sure where that observation gets us. I did say I'm on the fence, right?)

Reply:
I have tried the approach of decorating the functions with the HOSTDEVICE macro and selecting the implementation inside the function bodies, and it seems to work well. There is the added benefit that templates and composed functions work with relatively minor changes.

I'll push an updated version of fmath.h (and platform.h with the macro definition) once we have had a chance to review and test the changes.

Reply:
@timgrant was right, it needed the full separation he originally had. Either one will work when using clang to generate PTX, as he was doing here. But nvcc is much pickier, so when you are using fmath.h in a full nvcc-compiled Cuda program, you need his original approach. Sorry to make you do the extra work. I have refactored with this in mind in #1896.